<?xml version="1.0"?>
<launch>

  <!-- start camera synchronizer-->
	<group ns="camera">
		<group ns="color_left">
  		<node name="prosilica_driver" pkg="prosilica_camera" type="prosilica_node" output="screen">
    		<param name="ip_address" type="str" value="169.254.236.139"/>
    		<param name="trigger_mode" type="str" value="streaming"/>
    		<remap from="camera/image_raw" to="color_raw_data" />
    		<remap from="camera/camera_info" to="camera_info" />
    		<!--include file="$(find prosilica_camera)/cam_settings.xml"/-->
  		</node>
  		<node pkg="cob_camera_sensors" type="camera_synchronizer" name="camera_synchronizer" ns="sync" cwd="node" respawn="false" output="screen">
				<remap from="/camera/color_left/sync/image_non_sync" to="/image_non_sync"/>
				<!--remap from="image_non_sync" to="/camera/color_left/color_raw_data"/-->
				<!--remap from="image_sync" to="color_raw_data_sync"/-->
				<!--remap from="camera_info" to="camera_info_sync"/-->
  		</node>
		</group>
		<group ns="color_right">
  		<node name="prosilica_driver" pkg="prosilica_camera" type="prosilica_node" output="screen">
    		<param name="ip_address" type="str" value="169.254.47.10"/>
    		<param name="trigger_mode" type="str" value="streaming"/>
    		<remap from="camera/image_raw" to="color_raw_data" />
    		<remap from="camera/camera_info" to="camera_info" />
    		<!--include file="$(find prosilica_camera)/cam_settings.xml"/-->
  		</node>
  		<node pkg="cob_camera_sensors" type="camera_synchronizer" name="camera_synchronizer" cwd="node" respawn="false" output="screen">
				<remap from="image_non_sync" to="color_raw_data"/>
				<remap from="image_sync" to="color_raw_data_sync"/>
  		</node>
		</group>
		<group ns="tof">
  		<node pkg="cob_camera_sensors" type="tof_camera" name="tof_camera" cwd="node" respawn="false" output="screen">
				<remap from="grey_tof_data" to="tof_grey_data"/>
				<remap from="xyz_tof_data" to="tof_xyz_data"/>
  			<param name="configuration_files" value="../../common/files/cob3-1/"/>
  			<param name="camera_index" value="0"/>
  			<param name="tof_camera_type" value="CAM_SWISSRANGER"/>
			</node>
  		<node pkg="cob_camera_sensors" type="camera_synchronizer" name="camera_synchronizer_xyz" cwd="node" respawn="false" output="screen">
				<remap from="image_non_sync" to="tof_xyz_data"/>
				<remap from="image_sync" to="tof_xyz_data_sync"/>
  		</node>
  		<node pkg="cob_camera_sensors" type="camera_synchronizer" name="camera_synchronizer_grey" cwd="node" respawn="false" output="screen">
				<remap from="image_non_sync" to="tof_grey_data"/>
				<remap from="image_sync" to="tof_grey_data_sync"/>
  		</node>
		</group>
	</group>
</launch>
